### PR TITLE
object detection api proposal

### PR DESCRIPTION
**Pull request by KingBrewer**
To become rich one day

### CODE REVIEW

File: object-detections-api.yaml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+          in: formData
+          required: true
+          type: string
+          enum: ["800x600", "640x480"]
+        - name: imageFormat
+          description: allowed image formats
+          in: formData
+          type: string
+          required: true
+          enum: ["jpeg", "png"]
```
Comment (edholland) [inline object-detections-api.yaml]: bmp only plz

File: object-detections-api.yaml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+          schema:
+            $ref: '#/definitions/ApiResponse'
+        '405':
+          description: "Invalid input parameters"
+        '415':
+          description: "Unsupported media type. Only JPEG/PNG image formats are supported"
+        '503': 
+          description: "Service Unavailable - Neural Network's meltdown"
+        '504':
+          description: "Takes too much time to process an image. Back-off and try again later"
```
Comment (edholland) [inline object-detections-api.yaml]: what should be considered too much time? will back off help?

File: object-detections-api.yaml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+definitions:
+  DetectedObject:
+    type: string
+    description: classification of detected object
+    enum: ["person", "car", "animal"]
```
Comment (edholland) [inline object-detections-api.yaml]: lets stick with just person for mvp

File: object-detections-api.yaml [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+          description: "Invalid input parameters"
+        '415':
+          description: "Unsupported media type. Only JPEG/PNG image formats are supported"
+        '503': 
+          description: "Service Unavailable - Neural Network's meltdown"
+        '504':
+          description: "Takes too much time to process an image. Back-off and try again later"
+definitions:
+  DetectedObject:
+    type: string
```
Comment (edholland) [inline object-detections-api.yaml]: lets make this object, and include probability in it

File: object-detections-api.yaml (hunk 1 above)
Comment (KingBrewer) [reply object-detections-api.yaml]: as discussed offline, we'll stick with `jpeg` type only

File: object-detections-api.yaml (hunk 2 above)
Comment (KingBrewer) [reply object-detections-api.yaml]: That was proposal for you to have an ability to throttle consumer's requests. If :noidea: what do to with that on your side - let's just drop it. Please make your call @edholland

File: object-detections-api.yaml (hunk 3 above)
Comment (KingBrewer) [reply object-detections-api.yaml]: ack. btw that was populated intentionally for informational purposes.

File: object-detections-api.yaml (hunk 4 above)
Comment (KingBrewer) [reply object-detections-api.yaml]: awesome

File: object-detections-api.yaml (hunk 2 above)
Comment (edholland) [reply object-detections-api.yaml]: lets just apply natural back-pressure for now ;)

**Comment by KingBrewer** [comment]
erm... test 1.2.3.

**Comment by KingBrewer** [comment]
@edholland done

File: object-detections-api.yaml [hunk 5]
```diff
@@ -0,0 +1,86 @@
+swagger: "2.0"
+info:
+  description: "Extremely Simple And Portable Objects Detection API"
+  version: "0.0.1"
+  title: "Embedded Objects Detector And Classifier"
+  termsOfService: "https://cardiffdevlunch.io/terms/"
```
Comment (edholland) [inline object-detections-api.yaml]: `s/\.io/.co.uk/g` plz

**Comment by KingBrewer** [comment]
@edholland done

**Pull request merged**